### DISCUSSION TITLE
Ring camera motion alerts

### DISCUSSION THREAD

**Discussion by rahulpdev** [tsightler/ring-mqtt#745] [Feature Requests]
My Ring doorbell, indoor cam and floodlight camera all have the option to toggle motion alerts. This is distinct from the option to toggle motion detection. Is it possible to add a feature to expose this toggle so that it can be operated in HA in the same way as the motion detection switch?

Although there is a discussion on this topic [here](https://github.com/tsightler/ring-mqtt/discussions/516) I believe it was incorrectly closed as the Ring Modes feature does NOT permit the user to set the motion alerts status. Switching the Ring Mode permits the user to set (i) motion detection, (ii) motion warnings i.e. the device speaks to the 'intruder' and (iii) live view, but it does not enable the user to set motion alerts status i.e. if motion detection is ON, the ability to select whether or not I receive alerts via the Ring app.

Can this request be (re)opened?

I guess for the time being I could (a) switch off all alerts in the Ring app and (ii) try to replicate the Ring app alerts behaviour in HA with a script that sends a phone notification and then using an automation that only triggers the script based on the mode of my HA alarm control panel...and attaching the snapshot to my notification? The downside to this approach is (1) it sounds like a lot of work! and (2) when I click on the HA notification I guess it could take me to the RTSP stream but I find the performance is much worse than the stream in the Ring app.

**Comment by tsightler** [comment]
Motion detection is a device-side setting while motion alerts are a client-side setting.  Basically, when you disable motion detection, you are changing a setting on the camera, so ring-mqtt can do this as well.  When you disable motion alerts, you are changing a setting on your local Ring client which controls whether that client (Android, iOS, Web, etc) generates notifications on that specific client device or not, thus ring-mqtt can't change this setting because it is local to your device.  Stated another way, if you have two separate devices with the Ring app installed, one can have motion alerts enabled while the other does not.

**Reply by brutus6** [reply]
@tsightler - Just reading this and wondered if there is a solution using smart alerts. Are smart alerts in motion settings client side or device side? I think they might be device side. I believe I can change them in the app separately to whether I enable alerts or not on my phone.

Could there be an MQTT control built that changes the smart alerts on and off for person, vehicle and/or motion?

Just a thought!
![Screenshot_20250602-124000](https://github.com/user-attachments/assets/19ab3c35-d4f4-4a2a-a0ce-cbd9a0d4df7d)

**Reply by tsightler** [reply]
Not really sure if I understand what solution you are looking for.  Smart Alerts is also a device global setting, if you configure smart alerts to, for example, not send vehicle alerts, then no clients will receive vehicle alerts.  As a general rule I don't recommend enabling smart alerts when using ring-mqtt unless you are 100% sure you don't want motion state changes on those events.

**Reply by brutus6** [reply]
Sorry probably didn't explain the context there. 

You said:

_"Smart Alerts is also a device global setting, if you configure smart alerts to, for example, not send vehicle alerts, then no clients will receive vehicle alerts."_  

That's exactly it...

... so if they are device side and not client side, I guess I can get the MQTT Ring in Home Assistant to turn on and off smart alerts for "Person", "Vehicle" and "Motion". Therefore I can leave motion alerts on on my phone (client side) and control the activation of alerts in general by using the device side smart alerts section in Home Assistant (getting around the problem)?

This relies on MQTT Ring exposing those API controls to the Home Assistant Integration as entities of course.

Use case: Create an automation in Home Assistant for when home or away.

So when home:

Trigger: When Ring Alarm disarmed

Actions:
Smart alerts Person set recording = "true" and set alerts = "false"
Smart alerts Vehicle set recording = "true" and set alerts = "false"
Smart alerts Motion set recording = "true" and set alerts = "false"

Then when away:

Trigger: When Ring Alarm armed

Actions:
Smart alerts Person set recording = "true" and set alerts = "true"
Smart alerts Vehicle set recording = "true" and set alerts = "true"
Smart alerts Motion set recording = "true" and set alerts = "true"

NB: In both cases, client side "Alerts" enabled and just left alone

**Reply by tsightler** [reply]
The ability to turn off motion detection is already in ring-mqtt, so what you describe is already possible, without Smart Alerts.  Turning off Smart Alerts quite literally turns off the detection of those events, so ring-mqtt won't be able to know those events either, thus what you describe is 100% identical to the existing behavior of simply turning of motion detection completely.

**Reply by brutus6** [reply]
Got it — thanks for explaining. I think where I was coming from is that (at least from my testing) Smart Alerts don't actually turn off motion detection itself — they just control what happens after motion is detected.

For example, I’ve got mine set to only notify for people, with vehicles and general motion turned off in Smart Alerts, but recording enabled for all of them. In the app, I only get person notifications (because motion alerts are enabled on my phone), but when I check the timeline, all the events — person, vehicle, and general motion — are still recorded and visible.

So motion is always being detected — Smart Alerts just decide whether that event triggers a notification or not (and whether to record, if you have recording toggled per event type). That’s why I was thinking if Smart Alerts could be exposed in HA via MQTT, I could potentially control that filtering via automations, without fully turning off motion detection.

Hopefully that makes a bit more sense where I was going with it!

**Reply by tsightler** [reply]
Sure, that's great, but that means ring-mqtt will also only show motion when a person is detected.  That's fine if that's what you want, but I don't want to get myself into a situation where I'm having to explain to people constantly that the reason ring-mqtt isn't detecting motion is because they are turning off alerts.

Take for example the use case of the OP, he's trying to figure out a way to disable alerts on his phone based on arming mode, so yes, you could configure smart alerts with an automation to disable all alerts when disarmed, and send all alerts with armed.  In theory, this is great, but there's a side effect, when the alarm is disarmed, this will also disable ring-mqtt's ability to receive these alerts, so ring-mqtt will no longer detect motion at all.  If you are using ring-mqtt for an automation to, for example, turn on lights, setting Smart Alerts will break this exactly because it is global for the device.

I'm not saying this is right or wrong, I'm saying I don't want to spend the rest of my life closing issues on this project for people who will refuse to read and understand this limitation.

**Reply by brutus6** [reply]
Okay. That makes sense. Fair enough. I just thought I had found a work around to the motion alert issue people are having with it being client side. 

I just need to try and work out why my instance of ring-mqtt is not showing motion for my cameras. As I could just that to trigger an alert. I will raise that as a bug though and give you my version numbers and cameras etc. Very odd. Motion says "Clear" in Home Assistant and has done for hours, since set up, despite lots of motion events.
